iceoryx_hoofs/test/moduletests/test_posix_thread.cpp

```diff
@@ -35,66 +32,73 @@ class Thread_test : public Test
 
     void SetUp()
     {
-        m_run = true;
-        m_thread = new std::thread(&Thread_test::threadFunc, this);
     }
 
     void TearDown()
     {
-        m_run = false;
-        m_thread->join();
-        delete m_thread;
     }
 
     ~Thread_test()
     {
     }
 
-    void threadFunc()
+    uint32_t test_function(const uint32_t val)
     {
-        while (m_run)
-        {
-            std::this_thread::sleep_for(std::chrono::milliseconds(10));
-        }
-    };
-
-    std::atomic_bool m_run{true};
-    std::thread* m_thread;
+        return val;
+    }
+
+    optional<Thread> sut;
 };
 
-#if !defined(__APPLE__)
+TEST_F(Thread_test, CreateThread)
+{
+    constexpr uint64_t MY_FAVORITE_UINT = 13;
+    function<void()> callable = [&] { EXPECT_THAT(test_function(MY_FAVORITE_UINT), Eq(MY_FAVORITE_UINT)); };
+    ASSERT_FALSE(ThreadBuilder().create(sut, callable).has_error());
```

Comment:
How do you make sure that the `callable` is called at all.
I would capture a `bool callbackWasCalled = false;` set it to true inside the callback and check this callback from the outside.
Something like:
```cxx
bool wasCallbackCalled = false;
function<void()> callable = [&] { EXPECT_THAT(test_function(MY_FAVORITE_UINT), Eq(MY_FAVORITE_UINT)); wasCallbackCalled = true; };
ASSERT_FALSE(ThreadBuilder().create(sut, callable).has_error());
sut.reset(); // join the thread so that we are sure that the callback is finished
EXPECT_TRUE(wasCallbackCalled);
```

Reply:
Why have the check in the thread at all? Just checking that it was called should be sufficient, shouldn't it?